Added section on immutable types (#4437)
Particular/docs.particular.net: nservicebus/serialization/index.md

The [Newtonsoft JSON Serializer](newtonsoft.md) provides an effective general-purpose serializer appropriate for most use cases based on the ubiquitous [Json.NET package](http://www.newtonsoft.com/json). The Newtonsoft package provides a good combination of compactness, human readability, and performance. Other serializers are supported in order to enable specific performance or integration requirements.

## Immutable messages / types

> It is possible to use immutable types as messages. NServiceBus does not restrict this but it depends on the serializer implementation if it supports deserializing to non public properties.

Reviewer: ```suggestion
It is possible to use immutable types as messages. NServiceBus does not restrict this; It depends on the chosen serializer implementation if it supports deserializing to non public properties and/or using non-default constructors to initialize types.
```

> NOTE: On the wire it makes no difference if you use mutable or immutable message types.

Reviewer: ```suggestion
NOTE: On the wire it makes no difference if mutable or immutable message types are used.
```

> The [Newtonsoft JSON Serializer](newtonsoft.md) supports immutable messages types and probably many other serializers too but we do not maintain 

Reviewer: ```suggestion
For example the [Newtonsoft JSON Serializer](newtonsoft.md) can be configured to support immutable messages types.
```
